src/Uno.UI/UI/Xaml/Media/Animation/Animators/DispatcherAnimator.skia.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Security.Authentication.ExtendedProtection;
+using Microsoft.UI.Dispatching;
+using Windows.UI.Core;
+
+namespace Microsoft.UI.Xaml.Media.Animation
+{
+	internal abstract class DispatcherAnimator<T> : CPUBoundAnimator<T> where T : struct
+	{
+		public const int DefaultFrameRate = 60;
+
+		private Stopwatch _watch = new Stopwatch();
+		private TimeSpan? _delay;
+
+		public DispatcherAnimator(T from, T to, int frameRate = 0)
+			: base(from, to)
+		{
+		}
+
+		protected override void EnableFrameReporting() => CompositionTarget.Rendering += OnTargetFrame;
+		protected override void DisableFrameReporting() => CompositionTarget.Rendering -= OnTargetFrame;
+
+		private void OnTargetFrame(object sender, object args)
+		{
+			if (_delay != null)
+			{
+				if (_watch.Elapsed < _delay)
+				{
+					return;
+				}
+				else
+				{
+					_delay = null;
+					_watch.Stop();
+					_watch.Reset();
+				}
+			}
```

Comment:
You don't need all of this, this check is already done in the `OnFrame` https://github.com/unoplatform/uno/blob/1c42f17c61aedd3a043fe84b893e3c648940d2ea/src/Uno.UI/UI/Xaml/Media/Animation/Animators/CPUBoundAnimator.cs#L151.